opts.go

```diff
@@ -47,3 +48,11 @@ func WithProcessRunner(cmd *exec.Cmd) Opt {
 		machine.cmd = cmd
 	}
 }
+
+// WithSnapshot will allow for the machine to start using a given snapshot.
+func WithSnapshot(memFilePath, snapshotPath string, opts ...LoadSnapshotOpt) StartOpt {
+	return func(m *Machine) {
+		m.Handlers.FcInit.AppendAfter("SetupKernelArgsHandler",
+			NewLoadSnapshotHandler(memFilePath, snapshotPath, opts...))
+	}
```

Comment:
1) Don't we need to remove existing handlers when you have the snapshot handler? For example, you cannot start Linux after loading a snapshot.

2) Is it possible to validate some of the options in the snapshot case? For example, the kernel image won't be loaded if the VM is started from a snapshot.